src/protocol/mod.rs

```diff
@@ -75,23 +77,22 @@ impl Substep for str {}
     derive(serde::Deserialize),
     serde(from = "&str")
 )]
-pub struct Step {
+pub struct GenericStep {
```

Comment:
GenericGate?  Or does that make it too hard to deal with (comments and such).

Reply:
DescriptiveStep? StringBackedStep? not sure what is the best name here and what would be the appropriate name for "efficient" step backed by an int

Reply:
strings
* DescriptiveStep
* StringStep
* StringBackedStep
* VerboseStep
* TextualStep

integers
* IntStep
* NumericStep
* CompactStep
* MemoryEfficientStep

Reply:
I vote for `DescriptiveStep` and `CompactStep`